src/openforms/registrations/contrib/objects_api/tests/test_template.py

```diff
@@ -305,3 +309,109 @@ def test_submission_with_objects_api_content_json_not_valid_json(self):
+    def test_object_nulls_regression(self):
+        submission = SubmissionFactory.from_components(
+            components_list=[
+                {
+                    "type": "radio",
+                    "key": "radio",
+                    "label": "Radio",
+                    "values": [
+                        {"label": "1", "value": "1"},
+                        {"label": "2", "value": "2"},
+                    ],
+                    "defaultValue": None,
+                    "validate": {"required": True},
+                    "openForms": {"dataSrc": "manual"},
+                },
+                {
+                    "type": "textfield",
+                    "key": "tekstveld",
+                    "label": "Tekstveld",
+                    "hidden": True,
+                    "validate": {"required": True},
+                    "conditional": {"eq": "1", "show": True, "when": "radio"},
+                    "defaultValue": None,
+                    "clearOnHide": True,
+                },
+                {
+                    "type": "currency",
+                    "currency": "EUR",
+                    "key": "bedrag",
+                    "label": "Bedrag",
+                    "hidden": True,
+                    "validate": {"required": True},
+                    "conditional": {"eq": "1", "show": True, "when": "radio"},
+                    "defaultValue": None,
+                    "clearOnHide": True,
+                },
+            ],
+            with_report=True,
+            submitted_data={"radio": "2"},
+            form_definition_kwargs={"slug": "stepwithnulls"},
+        )
+        self._add_submission_to_session(submission)
+        # complete the submission
+        endpoint = reverse("api:submission-complete", kwargs={"uuid": submission.uuid})
+        with self.captureOnCommitCallbacks(execute=True):
+            response = self.client.post(endpoint, {"privacy_policy_accepted": True})
+        assert response.status_code == status.HTTP_200_OK
+        config = ObjectsAPIConfig(
+            objects_service=ServiceFactory.build(),
+            drc_service=ServiceFactory.build(),
+            content_json="{% json_summary %}",
+        )
+        plugin = ObjectsAPIRegistration(PLUGIN_IDENTIFIER)
+        prefix = "openforms.registrations.contrib.objects_api"
+
+        with (
+            patch(
+                f"{prefix}.models.ObjectsAPIConfig.get_solo",
+                return_value=config,
+            ),
+            patch(
+                f"{prefix}.submission_registration.get_documents_client"
+            ) as mock_documents_client,
+            patch(f"{prefix}.plugin.get_objects_client") as mock_objects_client,
+        ):
+            _documents_client = (
+                mock_documents_client.return_value.__enter__.return_value
+            )
+            _documents_client.create_document.return_value = {
+                "url": "https://example.com/dummy"
+            }
```

Comment:
Can't we skip the mocking of the documents client, by setting `informatieobjecttype_submission_report` to `None` in the registration options?

Reply:
succeeded!